edx_exams/apps/api/serializers.py

```diff
@@ -40,6 +47,73 @@ class Meta:
         fields = ['name', 'verbose_name', 'lti_configuration_id']
 
 
+class CourseExamConfigurationWriteSerializer(serializers.ModelSerializer):
+    """
+    Serializer for writing to the CourseExamConfiguration model.
+
+    We have separate read and write serializers because the read serializer uses the name field of the provider instance
+    to populate the provider serializer field by using the source attribute. The write serializer, on the other hand,
+    accepts the name as the value for the provider field. If we use the same serializer, deserialized data gets
+    represented as {'provider': {'name': <provider>}}, which makes reading the data less clear.
+    """
+    provider = serializers.CharField(allow_null=True)
+    # The escalation_email is a nullable field, but we require that clients send us an escalation_email.
+    escalation_email = serializers.EmailField(allow_null=True, allow_blank=False, required=True)
+
+    class Meta:
+        model = CourseExamConfiguration
+        fields = ['provider', 'escalation_email']
+
+    def validate_provider(self, value):
+        """
+        Validate that the provider value corresponds to an existing ProctoringProvider. If the value is None,
+        the value is valid.
+        """
+        if value is not None:
+            try:
+                ProctoringProvider.objects.get(name=value)
+            # This exception is handled by the Django Rest Framework, so we don't want to use "raise from e" and risk
+            # getting the stack trace included in client facing errors.
+            except ProctoringProvider.DoesNotExist:
+                raise serializers.ValidationError('Proctoring provider does not exist.') from None
+            return value
+
+        return value
+
+    def validate(self, attrs):
+        """
+        Validate that escalalation_email is provided when the provider is provided and that it is None if the provider
+        is None.
+        """
+        provider_name = attrs.get('provider')
+        escalation_email = attrs.get('escalation_email')
+
+        if provider_name and not escalation_email:
+            raise serializers.ValidationError('Escalation email is a required field when provider is provided.')
+        if provider_name is None and escalation_email is not None:
+            raise serializers.ValidationError('Escalation email must be None when provider is None.')
```

Comment:
This is a new validation error. In the spirit of being more explicit, I've removed the behavior I had originally, which was to set the `escalation_email` to `None` when the `provider` is `None`, even when the `escalation_email` is not `None` (e.g. `{'provider': None, 'escalation_email': 'test@example.com'}`). The client should supply valid data or receive a 400 error. I don't want to get in the habit of guessing what the client's intentions are when the `escalation` is not valid.